platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -183,17 +236,38 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
                     
                     NSSize targetSize = NSMakeSize(self.options.size.width, self.options.size.height);
                     NSRect targetFrame = NSMakeRect(0, 0, targetSize.width, targetSize.height);
+                    CGRect logoImageRect = CGRectMake(MGLLogoImagePosition.x, MGLLogoImagePosition.y, logoImage.size.width, logoImage.size.height);
+                    CGPoint attributionTextPosition = CGPointMake(targetFrame.size.width - 10 - attributionBackgroundSize.width,
+                                                                  logoImageRect.origin.y + (logoImageRect.size.height / 2) - (attributionBackgroundSize.height / 2));
+                    
+                    
                     NSImage *compositedImage = nil;
                     NSImageRep *sourceImageRep = [sourceImage bestRepresentationForRect:targetFrame
                                                                                 context:nil
                                                                                   hints:nil];
                     compositedImage = [[NSImage alloc] initWithSize:targetSize];
                     
                     [compositedImage lockFocus];
+
                     [sourceImageRep drawInRect: targetFrame];
-                    [logoImage drawInRect:CGRectMake(MGLLogoImagePosition.x, MGLLogoImagePosition.y, logoImage.size.width,logoImage.size.height)];
+                    
+                    [logoImage drawInRect:logoImageRect];
+                    
+                    NSRect cropRect = NSMakeRect(attributionTextPosition.x, MGLLogoImagePosition.y + 1, attributionBackgroundSize.width, attributionBackgroundSize.height);
```

Comment:
What should we do if the logo and attribution text overlap? This can also be tail work, but I wonder if there’s a quick fix in that case, like bumping the logo up above the attribution text and left-aligning the attribution text. (Or would that be too unpredictable?)